latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/TemplateProcessor.java

```diff
@@ -43,24 +48,32 @@
 
     private static final String BLOCK = "block";
 
-    private TemplateManager templateManager;
+    private final TemplateManager templateManager;
 
-    private VelocityMacroFilter filter;
+    private final VelocityMacroFilter filter;
 
-    private Map<String, Object> latexBinding;
+    private final Map<String, Object> latexBinding;
+
+    private final UIExtensionSupplier uiExtensionSupplier;
+
+    private final BlockRenderer blockRenderer;
```

Comment:
Would be good to decide if this is our best practice or not. I know I don't use `final` and I don't think others do either. I'd be happy to learn that we should do it and put it in our best practice if that's the case. Maybe start a forum proposal?